src/main/java/com/coveros/selenified/application/WaitFor.java

```diff
@@ -216,19 +238,65 @@ public void location(double seconds, String location) {
+        String expected = "Title shows as <b>" + title + "</b>";
         double end = System.currentTimeMillis() + (seconds * 1000);
         try {
             WebDriverWait wait = new WebDriverWait(driver, (long) seconds);
             wait.until(ExpectedConditions.titleIs(title));
             double timetook = Math.min((seconds * 1000) - (end - System.currentTimeMillis()), seconds * 1000) / 1000;
-            file.recordAction(action, expected, WAITED + timetook + " seconds for the title to show as '" + title + "'",
+            file.recordAction(action, expected, WAITED + timetook + " seconds for the title to show as <b>" + title + "</b>",
                     Result.SUCCESS);
         } catch (TimeoutException e) {
             // No alert found in given time
             file.recordAction(action, expected,
-                    WAITING + seconds + " seconds, the title shows as '" + driver.getTitle() + "'", Result.FAILURE);
+                    WAITING + seconds + " seconds, the title shows as <b>" + driver.getTitle() + "</b>", Result.FAILURE);
+            file.addError();
+        }
+    }
+
+    /**
+     * Wait up to a specified time for text to be present on the page
+     *
+     * @param seconds - the number of seconds to wait
+     * @param text    - the text to wait for
+     */
+    public void textPresent(double seconds, String text) {
+        String action = UPTO + seconds + " seconds for page to contain text <i>" + text + "</i>";
+        String expected = "Page contains text <b>" + text + "</b>";
+        double end = System.currentTimeMillis() + (seconds * 1000);
+        while (!file.getApp().is().textPresent(text) && System.currentTimeMillis() < end) ;
+        double timetook = Math.min((seconds * 1000) - (end - System.currentTimeMillis()), seconds * 1000) / 1000;
+        if (file.getApp().is().textPresent(text)) {
+            file.recordAction(action, expected, WAITED + timetook + " seconds for the page to contain text <b>" + text + "</b>",
+                    Result.SUCCESS);
+        } else {
+            // No alert found in given time
+            file.recordAction(action, expected,
+                    WAITING + seconds + " seconds, the page does not show text <b>" + text + "</b>", Result.FAILURE);
+            file.addError();
+        }
+    }
+
+    /**
+     * Wait up to a specified time for text to not be present on the page
+     *
+     * @param seconds - the number of seconds to wait
+     * @param text    - the to wait to not be present
+     */
+    public void textNotPresent(double seconds, String text) {
+        String action = UPTO + seconds + " seconds for page to not contain text <i>" + text + "</i>";
+        String expected = "Page doesn't contain text <b>" + text + "</b>";
+        double end = System.currentTimeMillis() + (seconds * 1000);
+        while (file.getApp().is().textPresent(text) && System.currentTimeMillis() < end) ;
```

Comment:
should we have a reference to app in the class like the other classes in this package?

Reply:
What do you mean?

Reply:
Like Assert.java has an app, 
`public Assert(App app, OutputFile file) {
    this.app = app;
    this.file = file;
}`
Just for consistencies sake, and cleaner code.

Reply:
gotcha, makes sense